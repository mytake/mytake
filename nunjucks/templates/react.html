{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
