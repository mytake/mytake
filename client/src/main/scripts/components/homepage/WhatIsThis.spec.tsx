/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as React from "react";
import * as renderer from "react-test-renderer";
import WhatIsThis from "./WhatIsThis";

jest.mock("./HomeSection", () => ({
  __esModule: true,
  default: "HomeSection",
}));
jest.mock("./NGramLoader", () => ({
  __esModule: true,
  default: "NGramLoader",
}));
jest.mock("./AnimatedHeading", () => ({
  __esModule: true,
  default: "AnimatedHeading",
}));
jest.retryTimes(3);

test("WhatIsThis", async () => {
  const tree = renderer
    .create(<WhatIsThis  />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
