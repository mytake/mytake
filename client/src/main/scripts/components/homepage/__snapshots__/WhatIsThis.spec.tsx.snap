// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhatIsThis 1`] = `
<HomeSection>
  <h2
    className="home__h1 home__h1--animated"
  >
    <AnimatedHeading
      onFinishTyping={[Function]}
    />
    <NGramLoader
      searchQuery=""
    />
  </h2>
</HomeSection>
`;
